### PR TITLE
02 -- Fellowship of the String

### PR DESCRIPTION
**Pull request by kellymurray**
### Review
- [x] What _primitive types_ exist in JavaScript?
- [x] What is an Expression?
  - What is the simplest Expression?
  - What is an Operator? What does an Operator require?
  - What is the result of an Operation? Expression!
### The Wonderful World of JavaScript: REDUX
- [x] What is a Statement?
  - [x] What Operator do we use to denote a Statement?
  - [x] What other Operators can separate Statements?
  - [x] Can a Statement contain more than one Expression?
- [x] What other kinds of Statements are there?
- [x] What is the Rule of Sequential Operation?
- [x] Are there exceptions to that Rule?
### A little more about Control Flow
- [ ] How do I alter the Control Flow of a program?
  - How do I _draw_ a decision tree? How do I _write_ one?
  - What if I have more than one decision to make?
  - How do I draw that?
- [ ] What if I need to make the same decision again?
### All about Functions
- [x] What is a Function?
  - How is a Function defined? How is a defined Function invoked?
  - What is the input? How does a function accept input?
  - What is the output? How does a function provide output?
  - What are side-effects? How does a function create side-effects?
- [ ] Do I always have to write my own Functions?
  - [x] What functions are built-in to JavaScript?
    - `console.log`, `console.assert`, what else?
  - [ ] Is there any other way to define or invoke a Function?
  - [ ] What is an anonymous function? What is it used for?
    - examples with `setTimeout`, `setInterval`
- [ ] How does this relate to Control Flow and the Rule of Sequential Operation?
### Coding Kata: String Calculator
- [ ] What's better than `console.assert`? `chai`...!
- [ ] How do I use a library like `chai` in my programs?
- [ ] What's better than ugly `AssertionError` messages? `mocha`...!
- [ ] How do I use `mocha` in my programs?
  - `npm install mocha`
  - `node_modules/.bin/mocha <filename>`
  - Man, that's ugly, huh?
- [ ] Let's keep track of that...
  - `npm init` to create a `package.json` file...
  - set the `test` command to `mocha`
  - now we can run `npm test <filename>` instead!
### To the `git` cave, Robin! ! (~30m)
- [ ] What did we change? -- `git status`
- [ ] No, really. -- `git diff`
- [ ] Time to commit!
- [ ] Pushing to a Remote

### CODE REVIEW

File: 02--Fellowship-of-the-String/README.md [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+* Are there exceptions to that rule?
+
+### Reading, Writing, and Running JavaScript
+* How can I _read_ JavaScript files?
+  * in the Finder with QuickLook...
+  * from the command line with `less <filename>`
+  * in a code editor like Atom!
+* How do I _write_ JavaScript files?
+  * in the Finder... (boo)
+  * from the command like with `touch <filename>`
```
Comment (al-the-x) [inline 02--Fellowship-of-the-String/README.md:36]: > ...from the command _line_ with...

Damn these typos!

File: 02--Fellowship-of-the-String/README.md [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+  * back to that Atom...!
+* How do I _run_ JavaScript?
+  * in the browser?
+  * from the command line with `node <filename>`
+  * in the browser?
+* What if I don't have a file yet? Use the `node` REPL!
+
```
Comment (al-the-x) [inline 02--Fellowship-of-the-String/README.md:43]: I'll probably launch into the Basic Calculator example here for scaffolding...

File: 02--Fellowship-of-the-String/README.md [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+    * examples with `setTimeout`, `setInterval`
+* How does this relate to Control Flow and the Rule of Sequential Operation?
+
+### A little more about Control Flow
+
+* How do I alter the Control Flow of a program?
+  * How do I _draw_ a decision tree? How do I _write_ one?
+  * What if I have more than one decision to make?
+  * How do I draw that?
+* What if I need to make the same decision again?
```
Comment (al-the-x) [inline 02--Fellowship-of-the-String/README.md:64]: I'm going to cover Control Flow ahead of Functions, I think...

File: 02--Fellowship-of-the-String/README.md [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+
+### Coding Kata: String Calculator
```
Comment (al-the-x) [inline 02--Fellowship-of-the-String/README.md:66]: Some bullet points for this section:
- What's better than `console.assert`? `chai`...!
- How do I use a library like `chai` in my programs?
- What's better than ugly `AssertionError` messages? `mocha`...!
- How do I use `mocha` in my programs?
  - `npm install mocha`
  - `node_modules/.bin/mocha <filename>`
  - Man, that's ugly, huh?
- Let's keep track of that...
  - `npm init` to create a `package.json` file...
  - set the `test` command to `mocha`
  - now we can run `npm test <filename>` instead!

**Comment by kellymurray** [comment]
### BLOCKERS
1. Markdown (hashtags belong on Twitter)
2. Reading and writing JS
3. Pull requests are weird, and so is forking, and so is GitHub
4. Git (why you gotta be so hard??)
5. What are expressions??

**Comment by al-the-x** [comment]
### Three Little Things
1. Markdown. How U even?
2. Talk the talk, walk the walk...
3. GitHub PR, why U hate me?

**Comment by kellymurray** [comment]
It's ALIVE!

**Pull request merged**